Updated README.md [9925] (#18)
eProsima/Discovery-Server: README.md

This mechanism is based on a client-server discovery paradigm, i.e. the metatraffic (message exchange among
DDS DomainParticipants to identify each other) is managed by one or several server DomainParticipants (left figure), as
opposed to simple discovery (right figure), where metatraffic is exchanged using a message broadcast mechanism like an
IP multicast protocol.
Please, refer to
[Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/server_client.html) for
further information about the Discovery Server discovery mechanism.

![discovery diagrams][diagrams]

[diagrams]: resources/images/ds_uml.png

The Discovery Server tool is developed to ease Discovery Server discovery mechanism setup and testing.
The **Discovery Server tool documentation** can be found
[here](https://eprosima-discovery-server.readthedocs.io/en/latest/).

-   [Supported platforms](#supported-platforms)
-   [Installation Guide](#installation-guide)
-   [Usage](#usage)
-   [Example application](#example-application)
-   [Testing](#testing)
-   [Documentation](#documentation)
-   [Getting Help](#getting-help)

## Supported platforms

* Linux
* Windows

## Installation Guide

> In order to use the Discovery Server tool, its necessary to have a compatible version of

Reviewer: ```suggestion
In order to use the Discovery Server tool, it is necessary to have a compatible version of
```

# eProsima Discovery Server

> The [RTPS standard](http://www.omg.org/spec/DDSI-RTPS/2.3) specifies in section 8.5 a non-centralized, distributed

Reviewer 2: ```suggestion
The [RTPS standard](http://www.omg.org/spec/DDSI-RTPS/2.3) specifies in section 8.5 a non-centralized, distributed,
```

> simple discovery mechanism. This mechanism was devised to allow interoperability among independent

Reviewer 2: ```suggestion
simple discovery mechanism. This mechanism was devised to allow interoperability between independent
```

vendor-specific implementations but it is not expected to be optimal in every environment.

> There are several scenarios were the simple discovery mechanism is unsuitable or plainly cannot be

Reviewer 2: ```suggestion
There are several scenarios where the simple discovery mechanism is unsuitable, or it just simply cannot be
```

IP multicast protocol.
Please, refer to
[Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/server_client.html) for
further information about the Discovery Server discovery mechanism.

![discovery diagrams][diagrams]

[diagrams]: resources/images/ds_uml.png

The Discovery Server tool is developed to ease Discovery Server discovery mechanism setup and testing.
The **Discovery Server tool documentation** can be found
[here](https://eprosima-discovery-server.readthedocs.io/en/latest/).

-   [Supported platforms](#supported-platforms)
-   [Installation Guide](#installation-guide)
-   [Usage](#usage)
-   [Example application](#example-application)
-   [Testing](#testing)
-   [Documentation](#documentation)
-   [Getting Help](#getting-help)

## Supported platforms

* Linux
* Windows

## Installation Guide

In order to use the Discovery Server tool, it is necessary to have a compatible version of
[eProsima Fast DDS](https://eprosima-fast-rtps.readthedocs.io/en/latest/) installed (over release 2.0.2).
*Fast DDS* dependencies as tinyxml must be accessible, either because *Fast DDS* was build-installed defining

> THIRDPARTY=ON or because those libraries have been specifically installed.

Reviewer 2: ```suggestion
`THIRDPARTY=ON|FORCE` or because those libraries have been specifically installed.
```

| eProsima/foonathan_memory_vendor:  | https://github.com/eProsima/foonathan_memory_vendor.git |    master   |
| leethomason/tinyxml2:              | https://github.com/leethomason/tinyxml2.git             |    master   |

---

## Usage

### XML profiles

Each setting in *Fast DDS* can be configured through XML profiles. XML profiles allows to avoid tiresome
hard-coded settings within applications sources using XML configuration files.
The *Fast DDS* XML schema was duly updated to accommodate the new Discovery Server tool settings.

-   The participant profile ``builtin`` tag contains a ``discovery_config`` tag where all discovery-related info is
    gathered. This new tag contains the following new XML child elements:
	- ``<discoveryProtocol>``: specifies the discovery type through the ``DiscoveryProtocol_t`` enumeration.

> 	- ``<discoveryServersList>``: specifies the server or servers linked with a Client/Server.

Reviewer 2: ```suggestion
	- ``<discoveryServersList>``: specifies the server or servers to which a Client/Server connects.
```

	- ``<clientAnnouncementPeriod>``: specifies the time span between PDP metatraffic exchange.

An example of XML profiles for a client and a server is shown below.

> ```xml

Reviewer 2: comments in here look weird

</participant>
```

### Discovery Server executable

The discovery server binary (named after the pattern ``discovery-server-X.X.X(d)`` where ``X.X.X`` is the version

> number and the optional *d* denotes a debug builds) is set up from one XML profiles files passed as command-line arguments.

Reviewer 2: ```suggestion
number and the optional *d* denotes a debug build) is set up from one XML profiles files passed as command-line arguments.
```
